### PR TITLE
Allow Developers to Access Order Resource

### LINKED ISSUE #4: Allow developers to access the Order resource
Verbs to be supported

- [x] GET all
- [x] GET one 
- [x] POST
- [x] PUT
- [x] DELETE

User should be able to GET a list, and GET a single item.

### PR DESCRIPTION
**Pull request by sarahmorrisokeefe**
# Description

This pull request allows the developer to access the Order data in our database using GET, POST, PUT, and DELETE methods. The developer is able to GET all orders and GET single orders, POST a new order, PUT a different value for an existing order, and DELETE orders that are currently active (I.E. do not have a payment type). The DELETE method also deletes all products from the `order_products` table that have the same `order_id`. 

This pull request requires that you already have the database created. Please do the following:
1. `~ npm install`
2. Create a `.env` file that matches the `.env.example` file, with your specified PORT number. We used PORT 8080, so that's what it is in the examples.
3. `~ mkdir json`
4. `node db/build_data.js`
5. `node db/build_table.js`
6. `nodemon app.js`

The final step should start your server on the port you selected. You are now ready to test the API calls using an app like Postman.

Fixes #4 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**GET**
- To get all orders, use the `GET` method and go to: **http://localhost:8080/api/v1/orders/** - this should return a JSON file of all existing orders, both active and inactive
- To get a single order, use the `GET` method and go to: **http://localhost:8080/api/v1/orders/<ORDER ID NUMBER>** - you need to insert the order ID number you're looking to get at the end of the URL, and it should return a JSON file of just one order with that specific order ID

**POST**
- To add an order, use the `POST` method and go to: **http://localhost:8080/api/v1/orders with the order that you want to add in JSON format. Here is my example:
```
{
	"customer_id": 92,
	"payment_type": 26
}
```
- If you refresh the data, you should see a new line with an automatically generated `order_id`, the new `customer_id` of 92, and the new `payment_type` of 26

**PUT**
- To edit a current order, use the `PUT` method and go to **http://localhost:8080/api/v1/orders/<ORDER ID NUMBER>** and add the `order_id` you want to edit the content for. Here is my example:
```
http://localhost:8080/api/v1/orders/45

{
	"customer_id":65,
	"payment_type":26
}
```
- This will update the data so that the order with an `order_id` of 45, now has a `customer_id` of 65, and the same `payment_type` of 26

**DELETE**
- The delete method will only work on an order that has an existing `order_id`, and has a `payment_type` of `NULL`
- To test, use the `DELETE` method and go to **http://localhost:8080/api/v1/orders/<ORDER ID>**. You will need to put in the `order_id` in the end of the URL in order to access that specific order (using route params). If you put an `order_id` that has a `payment_type` of `null`, the order should delete from the database. If you put an `order_id` that is already complete and has a number for the `payment_type`, it will return an error: 
```
{
    "message": "Error error error!",
    "error": "You cannot delete an order that has been completed"
}
```

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [N/A] I have added tests that prove my fix is effective or that my feature works
- [N/A] New and existing unit tests pass locally with my changes
- [N/A] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: controllers/ordersCtrl.js [hunk 1]
```diff
@@ -42,12 +42,28 @@ module.exports.putOneOrder = (req, res, next) => {
...
 }
 
+
+
 // DELETE
 module.exports.deleteOneOrder = (req, res, next) => {
-    deleteOne(req.body)
-    .then(order => {
-        res.status(200).json(order)
+    getOneActive(req.params.id)
```
Comment (iiimosley) [inline controllers/ordersCtrl.js]: replace `getOneActive` with `getOne`

File: models/Order.js [hunk 2]
```diff
@@ -47,11 +47,29 @@ module.exports.putOne = ({id}, {customer_id, payment_type}) => {
...
+    })
+    // return new Promise((resolve, reject) => {
+    //     db.run(`DELETE FROM orders WHERE order_id=${order_id} AND product_type=NULL`, error => {
+    //         if (error) return reject(error);
+    //         resolve({id: this.lastID});
+    //     });
+    // });
 };
+
+module.exports.getOneActive = (id) => {
```
Comment (iiimosley) [inline models/Order.js]: model function can be removed with the invocation of `getOne`

**Review by iiimosley — CHANGES_REQUESTED** [review]
Just a few things here:
- I believe get `getOne` and `getOneActive` use the same functionality. To minimize our codebase, will you please replace any invocation of `getOneActive` with `getOne`  and remove `getOneActive` from `Order.js`?   (see line comments)
- Please specify in your documentation that the `order_id` needs to be passed into the route params of the URL for Delete functionality (ie `../api/v1/orders/13`)
- Could you copy+paste this excellent testing documentation of your pull request into the README?

Otherwise, it's set to jet! 🚀  Thank you!

File: controllers/ordersCtrl.js (hunk 1 above)
Comment (sarahmorrisokeefe) [reply controllers/ordersCtrl.js]: Thanks @iiimosley - working on this now!

**Comment by sarahmorrisokeefe** [comment]
@iiimosley thanks for reviewing this - just pushed up new code to fix these issues:
- Removed all `getOneActive` mentions and took out the function from the model in `Order.js`
- Added to the documentation (README) that the user needs to use route params and put in the `order_id` at the end of the URL
- Added PR documentation to the README

**Review by iiimosley — APPROVED** [review]

**Review by iiimosley — APPROVED** [review]
Looking excellent,  thanks Sarah!  🍰

**Review by iiimosley — COMMENTED** [review]
still good 👍

**Comment by CourtneyBrothers** [comment]
totes saucy

**Pull request merged**